js/main.js

```diff
@@ -17,21 +17,65 @@ const $form = document.querySelector('.form');
 function formSave(event) {
   event.preventDefault();
   const formData = {};
-  formData[$form.elements.title.name] = $form.elements.title.value;
-  formData[$form.elements.photoURL.name] = $form.elements.photoURL.value;
-  formData[$form.elements.notes.name] = $form.elements.notes.value;
-  formData.entryId = data.nextEntryId;
-  data.nextEntryId++;
-  data.entries.unshift(formData);
-  $entryImage.setAttribute('src', 'images/placeholder-image-square.jpg');
-  $form.reset();
-
-  // Updated functionality of issue#2: view the entires
-  const newEntry = renderEntry(formData);
-  const $ulEntryList = document.querySelector('ul.entry-list');
-  $ulEntryList.prepend(newEntry);
-  viewSwap('entries');
-  toggleNoEntries('off');
+  if (data.editing === null) {
+    // Perform standard functionality
+    formData[$form.elements.title.name] = $form.elements.title.value;
+    formData[$form.elements.photoURL.name] = $form.elements.photoURL.value;
+    formData[$form.elements.notes.name] = $form.elements.notes.value;
+    formData.entryId = data.nextEntryId;
+    data.nextEntryId++;
+    data.entries.unshift(formData);
+
+    // Reset the form
+    $entryImage.setAttribute('src', 'images/placeholder-image-square.jpg');
+    $form.reset();
+
+    // Updated functionality of issue#2: view the entires
+    const newEntry = renderEntry(formData);
+    const $ulEntryList = document.querySelector('ul.entry-list');
+    $ulEntryList.prepend(newEntry);
+    viewSwap('entries');
+    toggleNoEntries('off');
+  } else {
+    // Updated functionality of issue#3: edit the entry
+    // assign the entry id value from data.editing to formData
+    formData.entryId = data.editing.entryId;
+
+    // record the new value
+    formData[$form.elements.title.name] = $form.elements.title.value;
+    formData[$form.elements.photoURL.name] = $form.elements.photoURL.value;
+    formData[$form.elements.notes.name] = $form.elements.notes.value;
+
+    // replace the original element in the array with the new one
+    let editEntryIndex = 0;
+    for (let indx = 0; indx < data.entries.length; indx++) {
+      if (data.entries[indx].entryId === formData.entryId) {
+        editEntryIndex = indx;
+      }
+    }
+    data.entries.splice(editEntryIndex, 1, formData);
```

Comment:
Why not just move this inside of your `if` conditional in your loop? That way you don't even need `editEntryIndex`, you can just use `indx`

Reply:
Thank you for your suggestion, Shawn. I have edited the code, pushed again, and merged.